### DISCUSSION TITLE
HTTP POST (2) rule in payload

### DISCUSSION THREAD

**Discussion by baphael** [jertel/elastalert2#1190] [Q&A] [answered]
Hello 👋 
I'm a bit confused about how to customize the HTTP POST (2) payload...
What I'd like is quite simple : 
```
{
  "match": {
    {{ match_key_values }}
  },
  "rule": {
    {{ rule_key_values }}
  }
}
```

Is there a way to achieve something like this ?

Thanks in advance.

**Comment by jertel** [comment]
The payload manipulation is documented with examples: https://elastalert2.readthedocs.io/en/latest/ruletypes.html#http-post-2

**Reply by baphael** [reply]
Hello,
well I should have mentioned that I dug the doc before posting here...

What I'm trying to achieve is not documented (perhaps not doable 🤷‍♂️).

I saw that you (@jertel) wrote that "_All rule options are available from within the `rule` dict object_" in #1135. However, unless I misunderstood, I couldn't find any way to retrieve rule's spec within the alert (using `http_post` or `http_post2`). Not even the rule's name at least.

Thx.

**Reply by jertel** [reply, marked as answer]
The doc shows this example, which should get you going:

![image](https://github.com/jertel/elastalert2/assets/16639611/f886d209-0115-4aeb-9eff-2e1aa2f419f6)

You may need to iterate through the match key/values in order to explicitly define the desired output for those keys and values for whatever format you are trying to achieve.

It doesn't look like the rule configuration keys are available. You would need to set those into the match object via a custom Python [enhancement](https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html).
